frontend/src/components/Metrics/CustomMetrics.tsx

```diff
@@ -226,7 +258,7 @@ class CustomMetrics extends React.Component<Props, MetricsState> {
     const content = (
       <>
         {this.renderOptionsBar()}
-        {this.state.dashboard && dashboard}
+        {this.state.dashboard !== undefined ? dashboard : this.renderFetchMetrics('Loading metrics', 'Metrics will be loaded in a few seconds...')}
```

Comment:
Kiali seems to be somewhat inconsistent on how it handles loading states. The graph page shows "Loading Graph" text but the other metrics charts and the list pages (istio config, workload, etc) either show nothing or show text saying something like "No Traces" until the data loads. Personally I prefer the loading text as it at least gives you the expectation that something is happening and will load eventually.

Can you have an empty custom tab with no dashboard? Would you then see "Loading metrics" permanently? If you can have an empty custom tab with no dashboard then there would be three states: dashboard is undefined (no dashboard), loading metrics (dashboard is loading could still return empty and go back to no dashboard), dashboard is defined.

Reply:
Imho the message "No metrics" could be confusing, as it shows metrics right after.
When there was a problem it loading or showing custom dashboards, there is a error message in notifications:
![Screenshot from 2022-09-09 09-14-07](https://user-images.githubusercontent.com/604313/189296206-e8ff73cc-98fc-4243-9dc9-3e117e8b3219.png)
So it s clear, that metrics were loading but error happened.

Reply:
There's potentially 4 states then!

1. Loading - request for dashboard
2. Dashboard(s) exist - finished loading successfully
3. No dashboard - finished loading successfully but empty dashboard
4. Error in request - finished loading but had an error in the request

It looks like you actually can have the custom tab with no dashboard but this is probably an invalid state and even in the main branch the page is currently blank so state 3 we don't have to worry about and state 4 is handled by the message center and more clear now by having the "Loading message".
![Screenshot from 2022-09-09 11-01-22](https://user-images.githubusercontent.com/6226732/189381363-501a673d-fa69-420e-aec6-bed79403158a.png)